wlauto/core/plugin.py

```diff
@@ -652,54 +556,343 @@ def _install_module(self, module):
+        self.paths = paths or []
+        self.ignore_paths = ignore_paths or []
+        self.plugins = {}
+        self.kind_map = defaultdict(dict)
+        self.aliases = {}
+        self.global_param_aliases = {}
+        self._discover_from_packages(self.packages)
+        self._discover_from_paths(self.paths, self.ignore_paths)
+
+    def update(self, packages=None, paths=None, ignore_paths=None):
+        """ Load plugins from the specified paths/packages
+        without clearing or reloading existing plugin. """
+        if packages:
+            self.packages.extend(packages)
+            self._discover_from_packages(packages)
+        if paths:
+            self.paths.extend(paths)
+            self.ignore_paths.extend(ignore_paths or [])
+            self._discover_from_paths(paths, ignore_paths or [])
+
+    def clear(self):
+        """ Clear all discovered items. """
+        self.plugins = []
+        self.kind_map.clear()
+
+    def reload(self):
+        """ Clear all discovered items and re-run the discovery. """
+        self.clear()
+        self._discover_from_packages(self.packages)
+        self._discover_from_paths(self.paths, self.ignore_paths)
+
+    def get_plugin_class(self, name, kind=None):
+        """
+        Return the class for the specified plugin if found or raises ``ValueError``.
 
-    def initialize(self, context):
-        pass
+        """
+        name, _ = self.resolve_alias(name)
+        if kind is None:
+            try:
+                return self.plugins[name]
+            except KeyError:
+                raise NotFoundError('plugins {} not found.'.format(name))
+        if kind not in self.kind_map:
+            raise ValueError('Unknown plugin type: {}'.format(kind))
+        store = self.kind_map[kind]
+        if name not in store:
+            raise NotFoundError('plugins {} is not {} {}.'.format(name, get_article(kind), kind))
+        return store[name]
+
+    def get_plugin(self, name=None, kind=None, *args, **kwargs):
+        """
+        Return plugin of the specified kind with the specified name. Any
+        additional parameters will be passed to the plugin's __init__.
+
+        """
+        name, base_kwargs = self.resolve_alias(name)
+        kwargs = OrderedDict(chain(base_kwargs.iteritems(), kwargs.iteritems()))
+        cls = self.get_plugin_class(name, kind)
+        plugin = cls(*args, **kwargs)
+        return plugin
+
+    def get_default_config(self, name):
+        """
+        Returns the default configuration for the specified plugin name. The
+        name may be an alias, in which case, the returned config will be
+        augmented with appropriate alias overrides.
+
+        """
+        real_name, alias_config = self.resolve_alias(name)
+        base_default_config = self.get_plugin_class(real_name).get_default_config()
+        return merge_dicts(base_default_config, alias_config, list_duplicates='last', dict_type=OrderedDict)
```

Comment:
`merge_dicts` here should be replaced with the equivalent call to `merge_config_values`